src/sentry/remote_config/docs/api.md

```diff
@@ -53,105 +43,231 @@ Retrieve the project's configuration.
+      "name": "custom-name",
+      "sample_rate": 1.0,
+      "traces_sample_rate": 0.5,
+      "version": 1
     }
   }
   ```
 
-**If no existing configuration exists**
+## Configuration [/projects/<organization_id_or_slug>/<project_id_or_slug>/configurations/<configuration_id>/]
+
+### Get Configuration [GET]
+
+Retrieve a configuration.
 
-- Response 404
+- Response 200
 
-### Set Configuration [POST]
+  ```json
+  {
+    "data": {
+      "environment": null,
+      "id": "0b88ac27a7b444a6baeb312c0493aed5",
+      "name": null,
+      "sample_rate": 0,
+      "traces_sample_rate": 0.75,
+      "version": 1
+    }
+  }
+  ```
+
+### Update Configuration [PATCH]
 
-Set the project's configuration.
+Update a configuration. The version attribute is required for all update requests and must match the current server configuration in order to update else an error is returned.
 
 - Request
 
   ```json
   {
     "data": {
-      "features": [
-        {
-          "key": "hello",
-          "value": "world"
-        },
-        {
-          "key": "has_access",
-          "value": true
-        }
-      ],
-      "options": {
-        "sample_rate": 1.0,
-        "traces_sample_rate": 0.5
-      }
+      "sample_rate": 0,
+      "version": 2
     }
   }
   ```
 
-- Response 201
+- Response 202
 
   ```json
   {
     "data": {
-      "features": [
-        {
-          "key": "hello",
-          "value": "world"
-        },
-        {
-          "key": "has_access",
-          "value": true
-        }
-      ],
-      "options": {
-        "sample_rate": 1.0,
-        "traces_sample_rate": 0.5
-      }
+      "environment": "production",
+      "id": "0b88ac27a7b444a6baeb312c0493aed5",
+      "name": null,
+      "sample_rate": 0,
+      "traces_sample_rate": 0.5,
+      "version": 2
     }
   }
   ```
 
 ### Delete Configuration [DELETE]
 
-Delete the project's configuration.
+Delete a configuration.
 
 - Response 204
 
-## Configuration Proxy [/remote-config/projects/<project_id>/]
+## Features [/projects/<organization_id_or_slug>/<project_id_or_slug>/configuration/<configuration_id>/features/]
 
-Temporary configuration proxy resource.
+### Get Features [GET]
 
-### Get Configuration [GET]
+Retrieve configuration features.
 
-Fetch a project's configuration. Responses should be proxied exactly to the SDK.
+- Response 200
+
+  ```json
+  {
+    "data": [
+      {
+        "description": "A feature description.",
+        "is_enabled": true,
+        "key": "hello",
+        "value": 22.3
+      }
+    ]
+  }
+  ```
+
+### Create Feature [POST]
+
+Create a configuration feature.
+
+- Request
+
+  ```json
+  {
+    "data": {
+      "description": "Another key.",
+      "is_enabled": true,
+      "key": "other",
+      "value": "key"
+    }
+  }
+  ```
+
+- Response 201
+
+  ```json
+  {
+    "data": {
+      "description": "Another key.",
+      "is_enabled": true,
+      "key": "other",
+      "value": "key"
```

Comment:
We should be returning the `id` field with the response here, and with the other feature objects returned.